File: auth-service/src/container.ts
import { AuthService } from './service/auth';
import { Config } from './config/index';
import { UserService } from "./service/user"

export type Container = {
    config: Config;
    authService: AuthService;
    userService: UserService;
}

export function initContainer(cfg: Config): Container {
    const userService = new UserService(cfg.DB_FILENAME);
    return {
        authService: new AuthService(cfg, userService),
        userService: userService,
        config: cfg
    }
}

File: fetch-service/internal/app/container.go
package app

import (
	"github.com/bagusays/efishery-technical-test/internal/config"
	"github.com/bagusays/efishery-technical-test/internal/service"
	"github.com/bagusays/efishery-technical-test/internal/shared/integration/currency_converter"
	"github.com/bagusays/efishery-technical-test/internal/shared/integration/efishery"
)

// Container :nodoc:
type Container struct {
	Cfg             *config.DefaultConfig
	AuthService     *service.Auth
	ResourceService *service.Resource
}

// NewContainer :nodoc:
func NewContainer(cfg *config.DefaultConfig) *Container {
	currencyConverterClient := currency_converter.NewClient()
	efisheryClient := efishery.NewClient()
	return &Container{
		AuthService:     service.NewAuth(),
		ResourceService: service.NewResource(currencyConverterClient, efisheryClient),
		Cfg:             cfg,
	}
}
